include/lsst/afw/geom/Angle.h

```diff
@@ -3,241 +3,294 @@
+     *
+     * @return the squared distance between the two vectors
+     */
+    double toUnitSphereDistanceSquared() const noexcept;
+
+    /**
+     * Return the angle between two unit vectors.
+     *
+     * This angle is given by @f$ \sin(\theta/2) = |\vec{u} - \vec{v}|/2 @f$.
+     *
+     * @param d2 the squared distance between two unit vectors
+     * @return the angle between the two vectors
+     */
+    static Angle fromUnitSphereDistanceSquared(double d2) noexcept;
+
+    /**
+     * Wrap this angle to the range [0, 2&pi;).
+     *
+     * @returns an angle in the normalized interval.
+     *
+     * @exceptsafe Shall not throw exceptions.
+     *
+     * @warning The upper limit is only guaranteed for radians; the upper limit
+     * may be slightly squishy for other units, due to roundoff errors. Whether
+     * there are any violations is unknown; please update this comment if you
+     * can prove that the limits are or are not valid for all supported units.
+     */
+    Angle wrap() const noexcept;
+
+    /**
+     * Wrap this angle to the range [-&pi;, &pi;).
+     *
+     * @returns an angle in the normalized interval.
+     *
+     * @exceptsafe Shall not throw exceptions.
+     *
+     * @warning Exact limits are only guaranteed for radians; limits for other
+     * units may be slightly squishy, due to roundoff errors. Whether there are
+     * any violations is unknown; please update this comment if you can prove
+     * that the limits are or are not valid for all supported units.
+     */
+    Angle wrapCtr() const noexcept;
+
+    /**
+     * Wrap this angle to a value `x` such that -&pi; &le; `x - refAng` < &pi;.
+     *
+     * @param refAng reference angle to match
+     *
+     * @returns an angle in the custom normalized interval.
+     *
+     * @exceptsafe Shall not throw exceptions.
+     *
+     * @warning Exact limits are only guaranteed for radians; limits for other
+     * units may be slightly squishy due to roundoff errors. There are known
+     * violations that are demonstrated in testWrap in tests/angle.py.
+     */
+    Angle wrapNear(Angle const& refAng) const noexcept;
+
+    /**
+     * The signed difference between two Angles.
+     *
+     * @param other the angle to which this angle will be compared
+     * @return `*this - other`, wrapped to the range [-&pi;, &pi;)
+     *
+     * @exceptsafe Shall not throw exceptions.
+     */
+    Angle separation(Angle const& other) const noexcept;
+
+#define ANGLE_OPUP_TYPE(OP, TYPE)                \
+    Angle& operator OP(TYPE const& d) noexcept { \
+        _val OP d;                               \
+        return *this;                            \
     }
 
-ANGLE_OPUP_TYPE(*=, double)
-ANGLE_OPUP_TYPE(*=, int)
-ANGLE_OPUP_TYPE(+=, double)
-ANGLE_OPUP_TYPE(+=, int)
-ANGLE_OPUP_TYPE(-=, double)
-ANGLE_OPUP_TYPE(-=, int)
+    //@{
+    /// Multiply this angle by the given factor.
+    ANGLE_OPUP_TYPE(*=, double)
```

Comment:
I don't see anything wrong with the `*=` operator; 2.5 times an angle is a well-defined geometric quantity. I'd agree with you on, e.g., `Angle += double`, but that's [DM-9630](https://jira.lsstcorp.org/browse/DM-9630).